### ISSUE TITLE
Implement average_batch_loss for all losses

### ISSUE THREAD

**Issue by le1nux** [le1nux/mlgym#2]
only bce_with_logits_loss offers to average_batch_loss

nll loss, cross entropy loss etc. don't have that yet!!
